network.go

```diff
@@ -1658,15 +1661,23 @@ func (n *network) ResolveIP(ip string) string {
 
 	nwName := n.Name()
 
-	n.Lock()
-	defer n.Unlock()
-	svc, ok = sr.ipMap[ip]
-
-	if ok {
-		return svc + "." + nwName
+	nameList, ok := sr.ipMap.Get(ip)
```

Comment:
any reason why it is named as nameList here while the master patch is called elemSet ?